src/components/study-manager.js

```diff
@@ -248,17 +250,60 @@ const StudyCard = ({study, onClick}) => {
 };
 
 const StudyManager = ({onStudyClick}) => {
-
+    const  userManagerHelper = new UserManagerHelper();
     const dispatch = useDispatch();
 
+    function login() {
+        userManagerHelper.getUserManagerInstance().signinRedirect();
+    }
+
+    function  getUser() {
+        userManagerHelper.getUserManagerInstance().getUser().then(user => {
+            if (user) {
+                dispatch(setLoggedUser(user));
+                console.debug('User has been successfully loaded from store.');
+            } else {
+                console.debug('You are not logged in.');
+                login()
+            }
+        });
+    }
+
+    function  renewToken()  {
+        userManagerHelper
+            .getUserManagerInstance()
+            .signinSilent()
+            .then(user => {
+                console.debug('Token has been sucessfully renewed. :-)');
+                getUser();
+            })
+            .catch(error => {
+                console.debug(error);
+            });
+    }
+
+    function  logout() {
+        userManagerHelper.getUserManagerInstance().signoutRedirect();
+    }
+
     useEffect(() => {
+        getUser();
         fetchStudies()
             .then(studies => {
                 dispatch(loadStudiesSuccess(studies));
             });
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    useEffect(() => {
```

Comment:
When you move the login/logout code elsewhere, you don't need this and the logout prop